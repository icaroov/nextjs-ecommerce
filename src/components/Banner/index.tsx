import Button from 'components/Button'
import Ribbon, { RibbonSizes, RibbonColors } from 'components/Ribbon'

import * as Styled from './styles'

export type BannerProps = {
  img: string
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string

  ribbon?: React.ReactNode
  ribbonSize?: RibbonSizes
  ribbonColor?: RibbonColors
}

const Banner = ({
  title,
  subtitle,
  img,
  buttonLabel,
  buttonLink,
  ribbon,
  ribbonSize = 'normal',
  ribbonColor = 'primary'
}: BannerProps) => (
  <Styled.Container>
    {!!ribbon && (
      <Ribbon size={ribbonSize} color={ribbonColor}>
        {ribbon}
      </Ribbon>
    )}
    <Styled.Image src={img} role="img" aria-label={title} />

    <Styled.Caption>
      <Styled.Title>{title}</Styled.Title>
      <Styled.Subtitle dangerouslySetInnerHTML={{ __html: subtitle }} />
      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </Styled.Caption>
  </Styled.Container>
)

export default Banner
