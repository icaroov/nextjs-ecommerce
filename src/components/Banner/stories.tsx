import { Story, Meta } from '@storybook/react/types-6-0'
import Banner, { BannerProps } from '.'

export default {
  title: 'Banner',
  component: Banner,
  args: {
    img:
      'https://sm.ign.com/ign_br/screenshot/default/world-of-warcraft-classic-onyxia_y46q.jpg',
    title: 'World of Warcraft',
    subtitle: '<p>Play the new <strong>expansion</strong> right now!',
    buttonLabel: 'Buy Now',
    buttonLink: '/games/world-of-warcraft'
  },
  argTypes: {
    ribbon: {
      type: 'string'
    }
  },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: Story<BannerProps> = args => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Banner {...args} />
  </div>
)

export const WithRibbon: Story<BannerProps> = args => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Banner {...args} />
  </div>
)

WithRibbon.args = {
  ribbon: '20% OFF',
  ribbonSize: 'normal',
  ribbonColor: 'primary'
}
