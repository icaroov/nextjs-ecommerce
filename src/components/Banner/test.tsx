import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Banner from '.'

const props = {
  title: 'World of Warcraft',
  subtitle: 'Play the new expansion right now!',
  buttonLabel: 'Buy now',
  buttonLink: '/games/world-of-warcraft',
  img:
    'https://sm.ign.com/ign_br/screenshot/default/world-of-warcraft-classic-onyxia_y46q.jpg'
}

describe('<Banner />', () => {
  it('should render the banner correctly', () => {
    const { container } = renderWithTheme(<Banner {...props} />)

    expect(screen.getByRole('img')).toHaveAttribute(
      'src',
      'https://sm.ign.com/ign_br/screenshot/default/world-of-warcraft-classic-onyxia_y46q.jpg'
    )

    expect(
      screen.getByRole('heading', { name: /world of warcraft/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', {
        name: /Play the new expansion right now!/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('link', {
        name: /buy now/i
      })
    ).toHaveAttribute('href', '/games/world-of-warcraft')

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a ribbon', () => {
    renderWithTheme(
      <Banner
        {...props}
        ribbon="My ribbon"
        ribbonSize="small"
        ribbonColor="secondary"
      />
    )

    const ribbon = screen.getByText(/my ribbon/i)

    expect(ribbon).toBeInTheDocument()
    expect(ribbon).toHaveStyle({ backgroundColor: '#3CD3C1' })
    expect(ribbon).toHaveStyle({ height: '2.6rem', fontSize: '1.2rem' })
  })
})
