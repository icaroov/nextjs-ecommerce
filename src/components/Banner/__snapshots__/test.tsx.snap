// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render the banner correctly 1`] = `
.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  padding: 0.8rem;
  border-radius: 0.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FAFAFA;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  cursor: pointer;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-size: cover;
  background-image: url(https://sm.ign.com/ign_br/screenshot/default/world-of-warcraft-classic-onyxia_y46q.jpg);
  background-position: center center;
  background-color: #EAEAEA;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  margin-bottom: 1.6rem;
  font-size: 1.4rem;
  font-weight: 400;
  color: #FAFAFA;
}

.c4 strong {
  color: #F231A5;
  font-weight: 600;
}

@media (max-width:1170px) {
  .c0 .sc-gsTCUz {
    right: 0;
  }

  .c0 .sc-gsTCUz::before {
    display: none;
  }
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
  }
}

@media (min-width:768px) {
  .c1 {
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c2 {
    position: absolute;
    left: 0;
    bottom: 0;
    padding: 4.0rem;
    border-radius: 0 0 0.4rem 0.4rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <div
    aria-label="World of Warcraft"
    class="c1"
    role="img"
    src="https://sm.ign.com/ign_br/screenshot/default/world-of-warcraft-classic-onyxia_y46q.jpg"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      World of Warcraft
    </h2>
    <h3
      class="c4"
    >
      Play the new expansion right now!
    </h3>
    <a
      class="c5"
      href="/games/world-of-warcraft"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</main>
`;
